lib/ox/element.rb

```diff
@@ -150,7 +150,18 @@ def locate(path)
     # _raise_ [NoMethodError] if no match is found
     def method_missing(id, *args, &block)
       has_some = false
-      ids = id.to_s
+      ids = id.to_s.sub('_', ':') # allows easy API with namespaces
+      
+      if ids[-1] == '!' # return array of elements matching ids
```

Comment:
I've thought about that, but there's no characters beyond `?` and `!` that I know I can use, and I did try with an extra parameter (using `:each` instead of number), but that got me code like `doc.root.element(:each).each{...}` which just didn't look right to me. The prettiest would be `doc.root.element.each` I think, but for that a node would have to have access to its parent.